_posts/2019-04-01-lecture-20.md

```diff
@@ -0,0 +1,639 @@
+\end{aligned}
+$$
+
+This is similar to the relationship between the Q-function and V-function in standard RL. Execpt that in the first equation, a log-integral-exponential approximates a max (also called "soft-max"). And in the second equation, the log-expectation-exponential is similar to a log-integral-exponential which is similar to a soft-max.
+
+Let us carefully consider the recursive equation for $Q_t(s_t, a_t)$. For the case of deterministic dynamics, the expectation over $s_{t+1}$ reduces to a single evaluation over $s_{t+1}$, and we get $Q_t(s_t, a_t) = r(s_t, a_t) + V_{t+1}(s_{t+1})$. This is the same equation for deterministic dynamics of standard RL.
+
+However, in general for stochastic dynamics, we end up taking the approximate max of $V(s_{t+1})$ over all $s_{t+1}$. That is, we end up being too "optimistic" about the transition. This is problematic because this results in learning risk-seeking behaviors where actions are chosen that may rarely lead to some states that are associated with high reward. We will see in the next section how such optimistic behavior can be addressed.
+
+### Optimal Policy
+
+In the homework, we will show that the optimal policy is given by
+
+$$
+\pi(a_t \mid s_t) = \frac{\beta_t(s_t, a_t)}{\beta_t(s_t)} = \exp \Big( Q_t(s_t, a_t) - V(s_t) \Big) = \exp(A_t(s_t, a_t)))
+$$
+
+where $A_t(s_t, a_t)$ is known as the "advantage" function, since it denotes the advantage of taking action $a_t$ with respect to the "average" action for that state. There are two advantages of this form -
+
+- It has a natural interpretation, that better actions (which have high advantage) are more probable.
+- It facilitates random tie-breaking: if two actions have the same advantage - since the policy is stochastic, it samples them equally likely. Hence, this provides an automatic tie-breaking mechanism.
+- We can add a temperature parameter inside the exponent, which can either focus on sharp peaks (converging to a deterministic policy), or sample more uniformly (converging to a completely randomly policy).
+
+***
+
+## Control via Variational Inference
+
+Given this framework of inference for reinforcement learning with graphical models, one can ask the question about the relationship between the objective function the inference-basd policy is trying to optimize. In other words, what objective does inference optimize? We take a look at the KL divergence between trajectory distributions.
+
+In the case for deterministic dynamics, we can express the probability of an optimal trajectory $\tau$ as
+
+$$
+p(\tau) \propto \left[ p(\mathbf{s}_1) \prod_{t=1}^T p(\mathbf{s}_{t+1} | \mathbf{s}_t \mathbf{a}_t) \right] \exp \left( \sum_{t=1}^T r(\mathbf{s}_t, \mathbf{a}_t) \right).
+$$
+
+On the other hand, the inference-based policy produces the following probability of the trajectory $\tau$:
+
+$$
+\hat{p}(\tau) \propto \mathbf{1}[p(\tau) \neq 0] \prod_{t=1}^T \pi(\mathbf{a}_t | \mathbf{s}_t).
```

Comment:
there is an extra slash here?

Reply:
Sorry I don't think I understand what you mean. Could you please elaborate?

Reply:
Hmm, I don't understand either.

Reply:
I meant the $\neq$ symbol was not properly rendered here?

Reply:
I double checked and the $\neq$ symbol looks okay to me.

Reply:
> @siddancha, minor comment: please change editor's name to Hao.
> 
> @zhisbug, the notes look good to me. Shall we merger after the change above?

Changed editor's name to Hao.